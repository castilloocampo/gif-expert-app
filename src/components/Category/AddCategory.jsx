import React, { useState } from 'react';
import PropTypes from 'prop-types'

const AddCategory = ({ onSetCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = ({ target: { value } }) => {
    setInputValue(value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      onSetCategories(cats => [inputValue, ...cats]);
      // onSetCategories(871263);
      setInputValue("");
    }
  }
  return (
    <form aria-label="form" onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder='Ingrese bÃºsqueda...'
        value={inputValue}
        onChange={handleInputChange} />
    </form>
  );
}

AddCategory.propTypes = {
  onSetCategories: PropTypes.func.isRequired,
}

export default AddCategory;
