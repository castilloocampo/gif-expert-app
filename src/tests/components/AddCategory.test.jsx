import { fireEvent, render } from '@testing-library/react';
import AddCategory from '../../components/Category/AddCategory';
import { screen } from '@testing-library/react';

describe('AddCategory', () => {

  const obj = {
    onSetCategories: jest.fn()
  }

  test('should show correctly', () => {
    const { container } = render(<AddCategory {...obj} />);
    expect(container).toMatchSnapshot();
  });

  test('should change input textBox', () => {
    render(<AddCategory {...obj} />);
    const input = screen.getByRole("textbox");
    const value = "uno";
    fireEvent.change(input, { target: { value } });
    expect(input.getAttribute('value')).toBe(value);
  });

  test("should'n submit information", () => {
    render(<AddCategory {...obj} />);
    const form = screen.getByRole('form');
    const input = screen.getByRole('textbox');
    fireEvent.submit(form);
    fireEvent.change(input, { target: { value: "h2" } });
    fireEvent.submit(form);
    expect(obj.onSetCategories).not.toHaveBeenCalled();
  });

  test("should submit information", () => {
    render(<AddCategory {...obj} />);
    const form = screen.getByRole('form');
    const input = screen.getByRole('textbox');
    // simular inputChange
    fireEvent.change(input, { target: { value: "Mario" } });

    // simular submit
    fireEvent.submit(form);

    // onSetCategories se llamo
    expect(obj.onSetCategories).toHaveBeenCalledWith( expect.any(Function));

    // value se vaci√≥
    expect(input.getAttribute('value')).toBe('');
  });

});
