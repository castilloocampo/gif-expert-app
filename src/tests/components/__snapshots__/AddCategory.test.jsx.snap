// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should show correctly 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Ingrese bÃºsqueda..."
      type="text"
      value=""
    />
  </form>
</div>
`;
